internal/controller/dnsrecord_delegation_helper.go

```diff
@@ -60,16 +60,21 @@ func (r *DNSRecordDelegationHelper) createAuthoritativeRecordFor(ctx context.Con
 }
 
 func authoritativeRecordFor(rec v1alpha1.DNSRecord) *v1alpha1.DNSRecord {
+	rootHostHash := common.HashRootHost(rec.Spec.RootHost)
 	return &v1alpha1.DNSRecord{
 		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: "delegation-authoritative-record-",
-			Namespace:    rec.Namespace,
+			Name:      toAuthoritativeRecordName(rootHostHash),
+			Namespace: rec.Namespace,
 			Labels: map[string]string{
-				v1alpha1.DelegationAuthoritativeRecordLabel: common.HashRootHost(rec.Spec.RootHost),
```

Comment:
do we want to label these records and authoratative: true so listing authoratitive records only is possible?